nomos-blend/message/src/reward/session.rs

```diff
@@ -62,6 +70,21 @@ impl From<[u8; 64]> for SessionRandomness {
     }
 }
 
+const SESSION_RANDOMNESS_TAG: [u8; 27] = *b"BLEND_SESSION_RANDOMNESS_V1";
+
+impl SessionRandomness {
+    /// Derive the session randomness from the given session number and epoch
+    /// nonce.
+    #[must_use]
+    fn new(session_number: SessionNumber, epoch_nonce: ZkHash) -> Self {
+        let mut hasher = Blake2b512::new();
+        hasher.update(SESSION_RANDOMNESS_TAG);
```

Comment:
Do you think it makes more sense to call the upper `blake2b512` function, and give it a chained slice of the inputs? Then we would avoid repeating logic. As far as I can see, it would be the same logic.

Reply:
No we have `crate::crypto::blake2b512`, although it's not visible to the outside, since we are dealing with same-create operations, I think we could make it `pub(crate)`.

Reply:
Ah, thank you. bab848711327b62473f3a0627ac1b7bd966840c5 I refactored the `blake2b512` little bit, so that we don't need to do `concat()` or `chain()`. I think this version is more efficient.

Unfortunately, `Blake2b512::update()` accepts `impl AsRef<[u8]>`. So, we cannot directly pass a chained byte array to it.